scheduler/scheduler-api/src/main/java/org/ow2/proactive/scheduler/signal/SignalApi.java

```diff
@@ -56,7 +58,7 @@
      * @return true if the signal is successfully added to the list of job signals
      * @throws InvalidChannelException if an error occurred while reading in the signals channel
      */
-    boolean readyForSignal(String signal) throws InvalidChannelException;
+    boolean readyForSignal(String signal) throws InvalidChannelException, IOException, CompilationException;
```

Comment:
you could eventually throw a RuntimeException instead of these Exception which are not necessarily understandable by the user (same for all methods).